logserver/src/main/java/alluxio/logserver/AlluxioLog4jSocketNode.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.logserver;
+
+import alluxio.AlluxioRemoteLogFilter;
+import alluxio.util.io.PathUtils;
+
+import com.google.common.base.Preconditions;
+import org.apache.log4j.Hierarchy;
+import org.apache.log4j.Level;
+import org.apache.log4j.Logger;
+import org.apache.log4j.PropertyConfigurator;
+import org.apache.log4j.spi.LoggerRepository;
+import org.apache.log4j.spi.LoggingEvent;
+import org.apache.log4j.spi.RootLogger;
+
+import java.io.BufferedInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.net.Socket;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Properties;
+
+/**
+ * Reads {@link org.apache.log4j.spi.LoggingEvent} objects from remote logging
+ * clients and writes the objects to designated log files. For each logging client,
+ * logging server creates a {@link AlluxioLog4jSocketNode} object and starts
+ * a thread serving the logging requests of the client.
+ */
+public class AlluxioLog4jSocketNode implements Runnable {
+  private static final org.slf4j.Logger LOG =
+      org.slf4j.LoggerFactory.getLogger(AlluxioLog4jSocketNode.class);
+  private final String mBaseLogsDir;
+  private final Socket mSocket;
+
+  /**
+   * Constructor of {@link AlluxioLog4jSocketNode}.
+   *
+   * Callers construct AlluxioLog4jSocketNode instances, passing the ownership of the socket
+   * parameter. From now on, the AlluxioLog4jSocketNode is responsible for closing the socket.
+   *
+   * @param baseLogsDir base directory for logs
+   * @param socket client socket from which to read {@link org.apache.log4j.spi.LoggingEvent}
+   */
+  public AlluxioLog4jSocketNode(String baseLogsDir, Socket socket) {
+    mBaseLogsDir = Preconditions.checkNotNull(baseLogsDir,
+        "Base logs directory cannot be null.");
+    mSocket = Preconditions.checkNotNull(socket, "Client socket cannot be null");
+  }
+
+  @Override
+  public void run() {
+    try (ObjectInputStream objectInputStream = new ObjectInputStream(
+        new BufferedInputStream(mSocket.getInputStream()))) {
+      LoggerRepository hierarchy = null;
+      while (!Thread.currentThread().isInterrupted()) {
+        LoggingEvent event;
+        try {
+          event = (LoggingEvent) objectInputStream.readObject();
+        } catch (ClassNotFoundException e) {
+          throw new RuntimeException("Class of serialized object cannot be found.", e);
```

Comment:
this will quite the event loop when e.g. a object failed to read. is this what we expect?

Reply:
If `objectInputStream.readObject()` throws exceptions, we handle them in the catch block. In both `ClassNotFoundException` and `IOException`, it is hard to recover, or it is beyond the server's capability to recover. Therefore this serving thread will terminate.